#437 - footnotes

wilkus27 opened the pull request:
Issue https://github.com/openstax/cnx-recipes/issues/437.

wilkus27 commented:
@helenemccarron This is how it displays currently:

![image](https://user-images.githubusercontent.com/40228854/62708849-7ce43480-b9f4-11e9-9371-8bfabaaf1fe6.png)

I need to removed space between text and border. It's so big because in the content footnotes have title:

![image](https://user-images.githubusercontent.com/40228854/62709050-eb28f700-b9f4-11e9-937c-25dbe330ad87.png)

At this moment I removed it by 

![image](https://user-images.githubusercontent.com/40228854/62709141-17dd0e80-b9f5-11e9-8793-e82b41436de1.png)

But maybe better way will be to trash it by recipes.

wilkus27 commented:
Should I also style footnote calls?

![image](https://user-images.githubusercontent.com/40228854/62712461-28908300-b9fb-11e9-924a-7aa0f8920061.png)

Usually, it could be done by prince properties.
But in this case footnotes are created in the content.

This is the markup from `collection.assemble.xhtml`

![image](https://user-images.githubusercontent.com/40228854/62713508-26c7bf00-b9fd-11e9-9d61-a6120a736352.png)

So that's why I change content to `none` for `::fotnote-call` and `::footnote-marker`

![image](https://user-images.githubusercontent.com/40228854/62714227-6fcc4300-b9fe-11e9-928e-02912af24227.png)

Please let me know if I wrong understand it.

helenemccarron commented:
@wilkus27 Regarding the spacing, have you tried `display: none`, this should remove the entire element from the DOM as opposed to removing its content. You may want to research the difference between `display: none` and `visibility: none`, one removes it altogether from the dom, the other one just hides the content. `content:none` is rarely, if not never used in CSS to hide content. 

For your second question, the footnotes should be moved as part of the baking, isn't that the case? (your screenshot shows collection.assembled, not collection.baked. And yes, you should style the footnote marker (the link in the content) as well.

wilkus27 commented:
@helenemccarron Thank you for a hint with `display: none`. That's what I was looking for :) 

About second problem, I think I asked the wrong question. What I want to know is, is it a good solution that I removed footnote call and marker created by prince?

I was confused, because usually there is such markup:

```
Some text <span class="some-class">Content of footnote</span> some text.
```
And using `float: footnote` for class `some-class` we get footnote area at the end of the page with automatically created footnote markers and content of the footnote and also footnote call is created in the place where is the span with this class.

In our case footnote markers and calls are already created:

![image](https://user-images.githubusercontent.com/40228854/62783385-99e43a80-babc-11e9-8d68-66a9550e1e52.png)

![image](https://user-images.githubusercontent.com/40228854/62783456-c435f800-babc-11e9-8b55-ab749d1a72e6.png)

![image](https://user-images.githubusercontent.com/40228854/62783559-f9dae100-babc-11e9-864b-1c6c9a2496fd.png)

And content of footnotes is at the end of the module, so when I'm using `float: footnote` for a `[data-type="footnote-refs"]` prince moved the whole container to the footnote area, but it generates footnote call at the end of the module not in the text.

![image](https://user-images.githubusercontent.com/40228854/62785841-b59e0f80-bac1-11e9-9a40-f9035632180e.png)


So as I said, I removed them and styled the existing ones. 
Should I remove existing calls and markers and try to find a way to generate them by prince in proper place?

helenemccarron commented on styles/templates/template1/parts/_footnotes-settings.scss:
```diff
@@ -0,0 +1,51 @@
+        Container: (
+            color: (_ref: 'colorMap:::fontBodyColor'),
+            line-height: line-height(1),
+            font-family: (_ref: 'typography:::baseFont'),
```
I this necessary, the basefont should be inherited

wilkus27 replied:
Inherited font is IBM Plex Sans. In specs from https://github.com/openstax/cnx-recipes/issues/437 font in footnotes should be Alegreya.

helenemccarron commented on styles/templates/template1/parts/_footnotes-settings.scss:
```diff
@@ -0,0 +1,51 @@
+        Border: (
+            border-top-color: (_ref: 'colorMap:::fontBodyColor'),
+            border-top-width: not-implemented(0.5pt),
+            width: not_implemented(11rem),
```
can you use the newly implemented grid() function? One grid unit = 10px

wilkus27 replied:
After merged new changes I implemented h-grid.

helenemccarron commented on styles/templates/template1/parts/_footnotes-settings.scss:
```diff
@@ -0,0 +1,51 @@
+            border-top-color: (_ref: 'colorMap:::fontBodyColor'),
+            border-top-width: not-implemented(0.5pt),
+            width: not_implemented(11rem),
+            padding-bottom: not_implemented(1.8rem),
```
Can you use the newly implemented vertical-rhythm() function?

wilkus27 replied:
After merged new changes I implemented v-grid.

helenemccarron commented on styles/templates/template1/parts/_footnotes-settings.scss:
```diff
@@ -0,0 +1,51 @@
+        ),
+        List: (
+            list-style: none,
+            margin: not-implemented(0),
```
Can you please use the spacing functions? Also, in general, we avoid setting top margins and top padding, so it could be avoided, it would be great (this is a special occurance, though, since it's very unlikely a footnote would end up on top of page) but I thought I'd let you know for later :D

wilkus27 replied:
I wasn't sure which functions should I implement here. You meant h-grid and v-grid? 

I'm also wondering that maybe this could be removed after merge https://github.com/openstax/cnx-recipes/pull/1284/commits/9ce72737ee1e37e5204ea3c1491143a368d047bd. There margin and padding are specified for `ul` in `_reset.scss`.
